File: server/marketplace/src/main/java/com/dspajic/marketplace/entities/PostDetails.java
package com.dspajic.marketplace.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PostDetails {

    private Integer id;

    private String title;

    private String description;

    private BigDecimal price;

    private String currency;

    private Integer userId;

    private Product product;

    private Location location;
}

File: client/client/src/types/PostDetails.ts
export interface PostDetails {
  id: number;
  title: string;
  description: string;
  price: number;
  currency: string;
  userId: number;
  product: Product;
  location: Location;
}

export interface Product {
  id: number;
  name: string;
  brandId: number;
  subcategoryItemId: number;
}

export interface Location {
  id: number;
  addressLine1: string;
  addressLine2: string | null; 
  cityId: number;
  postalCode: string;
}